docs/docs/troubleshooting/hardware-issues.mdx

```diff
@@ -0,0 +1,219 @@
+:::
+
+</TabItem>
+<TabItem value="visual">
+Visual inspection is generally less reliable, but a lot easier than either of the other approaches. Generally, you want to be looking for solder bridges, exposed copper, broken PCB traces, lifted pads, and similar artifacts.
+
+For through-hole soldering specifically, the below infographic can be quite helpful.
+![A helpful soldering infographic. Source: Sparkfun](https://cdn.sparkfun.com/assets/c/d/a/a/9/523b1189757b7fb36e8b456b.jpg)
+
+To confirm that something you've identified is indeed a problem, you'll want to make use of a multimeter or our tester shield.
+:::tip
+The above image was taken from [Sparkfun's guide on soldering through-hole components](https://learn.sparkfun.com/tutorials/how-to-solder-through-hole-soldering), which may be of use if you are less experienced with soldering.
+:::
+
+</TabItem>
+</PinTabs>
+
+### Resolving Issues
+
+Many issues can be resolved by reworking the affected solder joints (Look up a tutorial on how to do so for your component). Sometimes a pin ends up broken though, in which case you will need to "bodge" (solder) a wire from a spare GPIO pin to the connection in need of a working pin.
+
+Once you have done so, you will need to adjust the `kscan` of your keyboard slightly. The recommended approach to doing so is via your `zmk-config`:
+
+1. Search through the files defining your board/shield for the definition of the keyboard's `kscan`. This is typically a `.dts`, `.dtsi`, or an `.overlay` file.
+2. Add a [phandle](https://docs.zephyrproject.org/latest/build/dts/phandles.html) to your `.keymap` pointing at your `kscan`, copying in the property of your `kscan` where the broken GPIO pin appears. Depending on your `kscan` driver this could be `input-gpios`, `row-gpios`, etc.
+3. Edit the node to replace the old GPIO key with the new. For example:
+
+```dts
+&kscan0 {
+    input-gpios
+        = <&pro_micro 7 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&pro_micro 6 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&pro_micro 9 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        ;
+};
+```
+
+could have the pin `&pro_micro 6` (D6 in the [Pro Micro pinout](../development/hardware-integration/new-shield.mdx#shield-overlays)) replaced with `&gpio0 8` (P0.08 for nRF MCUs).
+
+```dts title="<my_keyboard>.keymap"
+&kscan0 {
+    input-gpios
+        = <&pro_micro 7 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&gpio0 8 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        , <&pro_micro 9 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>
+        ;
+};
+```
+
+You can then build and flash using your config as usual.
+
+:::note
+If the affected pin is not in your `kscan`, you will want to copy over and overwrite the node that it is in instead.
+:::
+
+## Antenna
```

Comment:
```suggestion
## Wireless Connectivity
```
This seems a bit awkward to call antenna, what about this?

Reply:
Made it a bit longer even, to avoid overlap with the connectivity issues page.